qcodes/utils/helpers.py

```diff
@@ -666,6 +667,23 @@ def inner(**inner_kwargs):
     return inner
 
 
+def create_on_off_val_mapping(on_val=True, off_val=False):
+    """
+    Val map which maps readonable on/off inputs to the specified on/off_vals
+    to be sent to the intrument and if inverted maps these back to bools.
+    """
+    return OrderedDict([('On', on_val),
+                       ('ON', on_val),
+                       ('on', on_val),
+                       ('1', on_val),
+                       (True, on_val),
+                       ('Off', off_val),
+                       ('OFF', off_val),
+                       ('off', off_val),
+                       ('0', off_val),
+                       (False, off_val)])
```

Comment:
you state that the inverse gets mapped to booleans. Is it ensured just by the fact that the True/False values are the latter ones in the lists corresponding to on_val and off_val? I'd love to have this ensured in a more strict way.

the minimal thing that can be done to achieve this is to extract as a function the code that creates the inverse val_mapping inside the `Parameter`, say `create_inverse_val_mapping`, and add a test that ensures that `create_inverse_val_mapping(create_on_off_val_mapping(on_val='on_value', off_val='off_value')) == {True: 'on_value', False: 'off_value'}`. Do you think that would suffice, @QCoDeS/core ?

Reply:
Yes this seems reasonable to add a test for this behavior.  
The inverse val mapping does indeed just iterate the key value pairs
so the last pair will always overwrite any previous definition. I think a test is enough. 
My primary concern would be that someone adds a new pair to the dict to handle 'oN' or some such 
thing and does not realize that it should not be added to the end. That is prevented by a test.